bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java

```diff
@@ -297,7 +297,7 @@ public void accept(List<AbstractComponent<?>> discoveredComponentsList) {
 
     @Override
     protected void updateThingStatus(boolean messageReceived, boolean availabilityTopicsSeen) {
-        if (messageReceived || availabilityTopicsSeen) {
+        if (!messageReceived || availabilityTopicsSeen) {
```

Comment:
This is incorrect and needs to be reverted.

Reply:
There has been an ongoing discussion on this here:   https://community.openhab.org/t/travis-ci-build-for-org-openhab-binding-mqtt-homeassistant/93042, maybe the wrong place to have it.  With the current logic, without the change, at least an MQTT Homeassistant light will never go offline when the payload of the MQTT availability_topic changes to offline.  Perhaps the proper fix belongs someplace else.  Thoughts?